build.sbt

```diff
@@ -621,6 +622,18 @@ lazy val zioJson: ProjectMatrix = (projectMatrix in file("json/zio"))
   )
   .dependsOn(core)
 
+lazy val weePickleV1Json: ProjectMatrix = (projectMatrix in file("json/weepickle"))
+  .settings(commonSettings)
+  .settings(
+    name := "tapir-json-weepickle-v1",
```

Comment:
I'd expect this naming to fail [sbt-shading](https://github.com/rallyhealth/sbt-shading) since latest major version seems to be [v0](https://github.com/softwaremill/tapir/releases/tag/v0.20.0-M10). Speaking of which, if we want to apply shading here, it should be applied to `core` as well (along with enforcement via sbt-shading). Looks like the version scheme is currently [early-semver](https://github.com/softwaremill/tapir/blob/ec90b4603e74971a27859370b94272df143e7c87/build.sbt#L64) which I understand to mean that binary compatibility can be broken at any time. If we import two binary-incompatible versions with the same artifact name, we end up with dependency hell.

The other option would be to leave this unshaded to signal (correctly) that the standard dangers apply.

Reply:
if we reverted these changes, how might we deal with a theoretical v2 weepickle here?

Reply:
The shading checks enforce that a jar's version matches `v[0-9]+` in its name. For cases where a shaded jar name needs to reference some remote jar version, we usually omit the `v` prefix for the remote version so the checker ignores it. That'd be `tapir-json-weepickle1` to disambiguate from a hypothetical `tapir-json-weepickle2`.

Reply:
ah right... forgot about that!

Reply:
That said, we could just leave it as `tapir-json-weepickle` for now since there aren't any current plans for a weepickle-v2. There are some small warts I'd like to fix, but it's not worth the impact on downstream consumers to break binary compatibility.